pyanaconda/modules/subscription/subscription.py

```diff
@@ -242,5 +441,438 @@ def set_system_purpose_with_task(self, sysroot):
+        self.rhsm_config_proxy.Set("logging.subscription_manager", get_variant(Str, "DEBUG"), "")
+        self.rhsm_config_proxy.Set("logging.rhsm", get_variant(Str, "DEBUG"), "")
+        self.rhsm_config_proxy.Set("logging.rhsm.connection", get_variant(Str, "DEBUG"), "")
+        self.rhsm_config_proxy.Set("logging.rhsm-app", get_variant(Str, "DEBUG"), "")
+        self.rhsm_config_proxy.Set("logging.rhsm-app.rhsmd", get_variant(Str, "DEBUG"), "")
+
+    def _start_private_bus(self):
+        """Start the RHSM private DBus session.
+
+        This used by RHSM for secure credential passing.
+        """
+        log.debug("RHSM: starting RHSM private DBus session")
+        register_server_proxy = RHSM.get_proxy(RHSM_REGISTER_SERVER)
+        private_bus_address = register_server_proxy.Start("")
+        self._rhsm_private_bus_address = private_bus_address
+        log.debug("RHSM: RHSM private DBus session started")
+
+    def _apply_syspurpose_from_kickstart(self):
+        """Apply system purpose information from kickstart to the installation environment.
+
+        The token transfer installation task will then make sure to transfer the result,
+        so the system purpose installation task does not have to run in a such a case.
+        """
+        log.debug("RHSM: Applying system purpose data from instalation kickstart")
+        task = SystemPurposeConfigurationTask("/", self.role, self.sla, self.usage, self.addons)
+        task.run()
+        # set system purpose as applied once the task finishes running
+        self._set_is_system_purpose_applied(True)
+
+    # insights
+
+    @property
+    def connect_to_insights(self):
+        """Indicates if the target system should be connected to red Hat Insights.
+
+        :return: True to connect, False not to connect the target system to Insights
+        :rtype: bool
+        """
+        return self._connect_to_insights
+
+    @connect_to_insights.setter
+    def connect_to_insights(self, connect):
+        """Set if the target system should be connected to Red Hat Insights.
+
+        :param bool connect: set to True to connect, set to False not to connect
+        """
+        self._connect_to_insights = connect
+        self.connect_to_insights_changed.emit()
+        log.debug("Connect target system to Insights set to: %s", self._connect_to_insights)
+
+    # install time tasks
+
+    def install_with_tasks(self, sysroot):
+        # subscription token transfer
+        token_transfer_task = TransferSubscriptionTokensTask(sysroot, self.entitlement_attached)
+        token_transfer_task_path = self.publish_task(SUBSCRIPTION.namespace, token_transfer_task)
+
+        # insights
+        insights_connect_task = ConnectToInsightsTask(sysroot, self.connect_to_insights)
+        insights_connect_task_path = self.publish_task(SUBSCRIPTION.namespace, insights_connect_task)
+        return [token_transfer_task_path, insights_connect_task_path]
```

Comment:
The `subscription.py` is too big. It should definitely be broken into smaller pieces for better maintenance. First step would be to separate RHSM service connection to a separate file.

In general the module main service class should be as small as possible and should not contain almost no logic.

Reply:
Hmm, yeah, I think we should move the "helper" non API functions to a separate file in the module directory. Looking on the `master` Anaconda that is definitely farther module wise than what we have currently on RHEL, there are couple such helper files in some modules.